tests/e2e/transitions/public_form_transitions.spec.js

```diff
@@ -230,16 +230,35 @@ const expectTransitions = (infodoc, ...transitions) => {
   });
 };
 
+const processSMS = (settings) => {
+  let ids;
+  return utils.updateSettingsNative(settings)
+    .then(() => Promise.all([
+      apiUtils.getApiSmsChanges(messages),
+      utils.requestNative(getPostOpts('/api/sms', { messages: messages }))
+    ]))
+    .then(([ changes ]) => {
+      ids = changes.map(change => change.id);
+    })
+    .then(() => sentinelUtils.waitForSentinelNative(ids))
+    .then(() => Promise.all([
+      utils.getDocsNative(ids),
+      sentinelUtils.getInfoDocsNative(ids),
+      utils.getDocNative('person1'),
+      utils.getDocNative('person2')
+    ]));
```

Comment:
Having this variable be mutated down below makes it fragile; seeing how it is involved in asynchronous code makes that fragility scarier. I see that most of this code is preexisting functionality (and I know we are leaving most of these things on this pass) with only changes to introduce the "native" functions. Still, I think this one is worth doing an async/await conversion on. It's pretty small and most of the `then` blocks don't rely on data from the others. Please refactor this block to move from the promise-based approach to using async/await.

Reply:
I think we should chat about this one. This one registers a listener `apiUtils.getApiSmsChanges(messages)` and I'm not 100% on handling that with async/await. I had originally tried converting but ran into that and eventually flipped back to not be blocked.

Reply:
Gotcha. In the interest of time, that's enough to hold off from refactoring this now.